taxi/views.py

```diff
@@ -79,6 +80,17 @@ class CarDeleteView(LoginRequiredMixin, generic.DeleteView):
     success_url = reverse_lazy("taxi:car-list")
 
 
+class AddOrDeleteUser(LoginRequiredMixin, View):
+    def post(self, request, pk):
+        car = get_object_or_404(Car, pk=pk)
+        if car.drivers.filter(id=request.user.id).exists():
+            car.drivers.remove(request.user)
+        else:
+            car.drivers.add(request.user)
+
+        return redirect("taxi:car-detail", pk=pk)
```

Comment:
Ensure that the logic for adding or removing a user from a car's drivers is correct and handles edge cases, such as when a user is not authenticated or when the car does not exist. Consider adding error handling or user feedback for these scenarios.